internal/telemetry/telemetry.go

```diff
@@ -124,6 +124,13 @@ func Track(ctx context.Context, exec shared.InteractionType, fn func(ctx context
 	// Execute the provided function, capturing any error
 	err = fn(ctx, runEvent)
 
+	// Populate event with pull request env var (available only after run)
```

Comment:
I'm not super excited about having to move this outside `EnrichEventWithEnvironmentVariables` which otherwise was the perfect place to do this, but short of a more involved refactor this should do the trick.